ckan/tests/controllers/test_package.py

```diff
@@ -38,6 +39,43 @@ def test_form_renders(self):
         env, response = _get_package_new_page(app)
         assert_true('dataset-edit' in response.forms)
 
+    @helpers.change_config('ckan.auth.create_unowned_dataset', 'false')
+    def test_needs_organization_but_no_organizations_has_button(self,):
+        ''' Scenario: The settings say every dataset needs an organization
+        but there are no organizations. If the user is allowed to create an
+        organization they should be prompted to do so when they try to create
+        a new dataset'''
+        app = self._get_test_app()
+        sysadmin = factories.Sysadmin()
+
+        env = {'REMOTE_USER': sysadmin['name'].encode('ascii')}
+        response = app.get(
+            url=url_for(controller='package', action='new'),
+            extra_environ=env
+        )
+        assert 'dataset-edit' not in response.forms
+        assert url_for(controller='organization', action='new') in response
+
+    @patch('ckan.logic.auth.create.organization_create',
+           return_value={'success': False})
+    @helpers.change_config('ckan.auth.create_unowned_dataset', 'false')
+    def test_needs_organization_but_no_organizations_no_button(self, *args):
+        ''' Scenario: The settings say every dataset needs an organization
+        but there are no organizations. If the user is not allowed to create an
+        organization they should be told to ask the admin but no link should be
+        presented'''
+        app = self._get_test_app()
+        sysadmin = factories.Sysadmin()
```

Comment:
You definitely can't get into this situation without a custom `package_create` auth logic. I think it's fine to patch `package_create` to mimic the overriding from an extension as that would be essentially what it will cause, just add a comment explaining why you are doing it.

So mock `package_create` to return True + `@helpers.change_config('ckan.auth.create_unowned_dataset', 'false')` + `@helpers.change_config('ckan.auth.user_create_organizations', 'false')`